### PR TITLE
feat: add new input to allow include statements at the top of formula class

### PR DESCRIPTION
**Pull request by unfrgivn**
Following the official guidance [here](https://github.com/Homebrew/brew/blob/master/docs/Python-for-Formula-Authors.md) for building formulas that deploy python apps. Homebrew provides built-in helper methods for different languages so I have called this generically `formula_includes` to differentiate that they are inside the Formula (open to suggestions on naming). In order to use these, they must be included inside the Formula class, but outside any of the blocks that Justintime50/homebrew-releaser currently allows for. 

I have added a generic placeholder that sits in the global scope of the Formula class, immediately following the metadata properties. 

*Example formula.rb*
```ruby
class TestGenerateFormula < Formula
  desc "Tool to release scripts, binaries, and executables to github"
  homepage "https://github.com/Justintime50/test-generate-formula"
  ...

  include Language::Python::Virtualenv <-- HERE IS THE NEW TEMPLATE BLOCK

  depends_on "bash" => :build
  depends_on "gcc"

  def install
    ...
```

**TODO:** *I still need to add tests to this*

### CODE REVIEW

**Comment by unfrgivn** [comment]
Let me get your thoughts @Justintime50 before I do any further clean up on this.

**Comment by Justintime50** [comment]
Awesome! I'm for it. This will enable Python apps installable correct?

A few thoughts:
1. Based on the docs provided, it looks like homebrew wants that `includes` line as the first item in the formula even before the metadata properties followed by a newline, let's follow suit in case there is a `brew audit` rule about it so we stay compliant.
2. Out of curiosity, what other `includes` are possible?
3. The lint is failing because I bumped the dev deps inbetween your last PR. You'll want to `just clean install` and then `just lint-fix` to pull in the new flake8/mypy rules and fix them. It looks like it corrected them back to the old rules based on your locally installed deps which no longer match the main branch.

**Comment by unfrgivn** [comment]
Looks like Brew has Ruby packages for a few languages with some helper functions, https://rubydoc.brew.sh/Language.html.

Not my area of expertise, but I would assume you could include any valid Ruby package and then reference it inside in the formula.

**Comment by unfrgivn** [comment]
Brew audit actually failed with extra whitespace. Fixing that last issue now.

File: test/unit/test_formula.py [hunk 1]
```diff
@@ -684,3 +684,47 @@ def test_generate_formula_override_version():
     record_formula(formula_path, formula_filename, formula)
 
     assert '9.8.7' in formula
+
+
+def test_generate_formula_formula_includes():
+    """Tests that we generate the formula content correctly when all parameters are passed
```
Comment (Justintime50) [inline test/unit/test_formula.py:690]: suggestion: we don't need to test all params here since we have other tests that cover that. What we'd be interested in for this test is only if the include line makes it into the generated formula and it's position and structure passed `brew audit`.

File: test/unit/test_formula.py [hunk 2]
```diff
@@ -684,3 +684,47 @@ def test_generate_formula_override_version():
...
+    mock_repo_name = formula_filename.replace('_', '-').replace('.rb', '')
+    mock_tar_url = f'https://github.com/{USERNAME}/{mock_repo_name}/archive/refs/tags/v0.1.0.tar.gz'
+
+    repository = {
+        # We use a badly written description string here on purpose to test our formatting code, this includes:
+        # - starting with an article
+        # - punctuation
+        # - trailing whitespace
+        # - extra capitilization
+        'description': 'A tool to release... scripts, binaries, and executables to GitHub. ',
```
Comment (Justintime50) [inline test/unit/test_formula.py:705]: suggestion: same as above, we'd probably want instead to just use the generic `DESCRIPTION` const here to avoid duplicating this edge case test that is already tested elsewhere.

**Review by Justintime50 — APPROVED** [review]
Looks great! Small suggestion,

TLDR: the test could be even simpler to ensure we are only testing what we care about in this particular case since we have tests for those other items elsewhere.

**Comment by Justintime50** [comment]
v0.19.0 (or v1) shipped with the feature, thanks for adding that! 😃
